odoo/addons/base/models/ir_fields.py

```diff
@@ -439,10 +439,15 @@ def _str_to_one2many(self, model, field, records):
             # [{subfield:ref1},{subfield:ref2},{subfield:ref3}]
             records = ({subfield:item} for item in record[subfield].split(','))
 
-        def log(_, e):
-            if not isinstance(e, Warning):
-                raise e
-            warnings.append(e)
+        RelatedModel = self.env[field.comodel_name]
+        field_names = {name: field.string for name, field in RelatedModel._fields.items()}
+        def log(f, exception):
+            if not isinstance(exception, Warning):
+                args = [exception.args[0] % dict(field=("%(field)s/" + field_names[f]))]
+                if len(exception.args) > 1 and exception.args[1]:
+                    args.append(exception.args[1])
+                raise type(exception)(*args)
+            warnings.append(exception)
```

Comment:
What kind of black magic is this?
I need an explanation.

Reply:
@rco-odoo This is something similar to https://github.com/odoo/odoo/blob/master/odoo/models.py#L1126
Here what we did, if its a m2o inside o2m and if this log method is called due to import fail, we will have args something like:
```("No matching record found for name 'New' in field '%(field)s'", {'moreinfo': {'name': 'Possible Values', 'type': 'ir.actions.act_window', 'target': 'new', 'view_mode': 'tree,form', 'views': [(False, 'list'), (False, 'form')], 'context': {'create': False}, 'help': 'See all possible values', 'res_model': 'your model'}})```

So to get full display name we appended string of m2o field(of o2m) here in args[0], say for example if we importing sale order and if we have sale order line o2m in file which may have product which is not available in system then it will fail, so we want full string like No matching record found for "Order Line/Product"...

So the above code just appends extra m2o field's string, let me know if I miss something, do I need to check for specifically m2o field?

Reply:
Couldn't you just replace args itself (possibly args[0] in place but that's probably a tuple) and directly use `_fields[f].string`? I don't see the need to build a mapping here. 

Something along the lines of:
```python
if not isinstance(exception, Warning):
    current_field_name = self.env[field.comodel_name]._fields[f].string
    arg0 = exception.args[0] % {'field': '%(field)s/' + current_field_name}
    exception.args = (arg0, *exception.args)
    raise
```

Reply:
@xmo-odoo done